spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -2366,6 +2372,18 @@ def run_script(self, filename, wdir, args, debug, post_mortem,
             )
 
     # ---- For working directory and path management
+    @property
+    def current_working_directory(self):
+        """
+        Current working directory, as reported by the Working Directory plugin.
+        """
+        return self._current_working_directory
+
+    @current_working_directory.setter
+    def current_working_directory(self, new_value):
```

Comment:
Maybe call this one `set_current_working_directory` and use it to set it instead of changing the attribute